### PR TITLE
Mobile profile edit DESKTOP-1653

### PR DESCRIPTION
**Pull request by awendland**
### Mmmk, wut is dis?

Well let me tell you: this is an official Red Ryder, carbine action, two-hundred shot range model air rifle... or at least it was until Mrs. Parker told me I'd shoot my eye out. So I instead changed it to be a PR implementing the mobile version of the edit profile information screen.

_But wait folks, there's more!_

To complete the mundane task of a mobile edit profile screen, I had to rely on a little component called Text Input. But, you see, Text Input had some problems. Mainly, it was incompatible between mobile OSs in 98235 different ways when multiline was enabled; so my first task was the almost (so close 👌) Herculean endeavor of sculpting the implementation of `common-adapters/input.native.js` into a standardized-esque shape. And boy, was it a blast. Spinning up emulators and simulators, fixing a bug on one OS only to have a new one manifest on another, making hurumph noises to ensure that everyone in the offices knew I was frustrated, brewing tea to sample all the WeWork flavors while I still had the chance.... it was a walk through the depths of hell, but with the help of Spotify and Slack's #random I made it.

Now, the number of changes may not look like much, but behind each line is man(human?)-years of effort and experience crafting the perfect style call and pixel offset value. A complete paradigm change was implemented! A full 180 moving from `position: absolute` to leveraging the magical world of `display: flex`. Plus, there's a new compiler hidden in here somewhere that will make our React code 42% faster!

Needless to say, "LGTM" would be an understatement after code-reviewing this PR. It's beauty and functionality are unparralled by all but the best of JavaScript code (probably on par with some of the greats, like http://stackoverflow.com/a/34458158). Just to make my point abundantly clear, bask in the glory of these screenshots:

| iOS | iOS w/ Error |
|----|--------------|
| screen shot 2016-08-26 at 4 05 41 pm | screen shot 2016-08-26 at 4 05 29 pm | 

| Android | Almost-as-cool "wonder of the world" |
|---------|----------|
| screen shot 2016-08-26 at 4 05 57 pm | screen shot 2016-08-26 at 4 05 57 pm |

### Wow-wee! This is a great PR!

Colbert Yes

@keybase/react-hackers 

This will probably be my last PR as an intern. Thanks for an amazing summer, and I can't wait to start Keybase-ing all my friends (even my grandma) in the coming years 😄

### CODE REVIEW

**Comment by chromakode** [comment]
Alright, this is now up to date with master and working in my emulators. The auto sizing behavior is janky. In iOS I see a flicker when it resizes, and in Android the text aligns weird vertically as new lines are input (and doesn't scroll to the cursor as you add lines). From my survey of RN issue threads this appears to be a pretty recent addition, so we should probably defer on making it perfect and wait for this to stabilize upstream. Let's land this for now and revisit when polishing.

@keybase/react-hackers

File: shared/common-adapters/input.native.js [hunk 1]
```diff
@@ -55,65 +96,74 @@ class Input extends Component<void, Props, State> {
...
-  position: 'relative',
-  height: 21,
-  top: -12,
 }
 
-const errorText = {
-  textAlign: 'center',
-  position: 'absolute',
+const HorizontalLine = ({focused}) => <Box style={{
+  position: 'relative',
```
Comment (MarcoPolo) [inline shared/common-adapters/input.native.js]: Does this need this `position: 'relative'`?

Comment (chromakode) [inline shared/common-adapters/input.native.js]: I'll check!

**Comment by MarcoPolo** [comment]
👍

**Pull request merged**